style-test.py

```diff
@@ -0,0 +1,148 @@
+    # find first occurence of style-guide checks in proselintrc
+    for index, line in enumerate(buf):
+        if line.startswith("        \"style-guide."):
+            pos = index
+            break      
+
+    # copy over new checks to proselintrc from style-text.txt
+    with open(dest_prc, "w") as out_file:
+        for index, line in enumerate(buf):
+            if index == pos-1 and not line.endswith(",\n"):
+                line = line[0:line.rfind("\n")] + ",\n"
+            if index < pos:
+                out_file.write(line)
+        for test in tests:
+            out_file.write(test)                
+        out_file.write("    }\n")
+        out_file.write("}")
+
+
+def remove_ignored_lines(txt):
+    """Function to remove ignored lines from text"""
+    index = 0
+    while index<len(txt):
+        if "startignore" in txt[index]:
+            while index<len(txt) and "endignore" not in txt[index]:
+                txt[index]="ignore_line\n"
+                index = index+1
+        index = index+1 
+
+    f= open("temp.txt","w+")
+    for line in txt:
+        f.write(line)
+
+    txt = open("temp.txt", "r")
+    os.remove("temp.txt")
+
+    return txt            
+
+
+def run_checks(paths):
+
+    """Function to run checks on the docs."""
+    file_path = os.path.realpath(__file__)
+
+    # find path for all .rst files
+    search_path = file_path[0:file_path.rfind('/')]
+
+    # Make a list of paths to check for
+    path_list = []
+
+    # Add paths if provided by user
+    if paths:
+        search_path = search_path + "/"
+        path_list = [search_path + path for path in paths]
+
+    # Add all rst files if specific paths not provided by user
+    else:
+        for filename in glob.glob(os.path.join(search_path, '*.rst')):
+            path_list.append(filename)
+
+    for filename in path_list:        
+        
+        # read the file 
+        with open(filename, "r") as file:
+            txt = file.readlines()
+
+        # remove ignored lines
+        txt = remove_ignored_lines(txt)
+
+        # lint the text
+        errors = proselint.tools.lint(txt)
+        err_list = []
+
+        for e in errors:
+
+            # ignore tests for curly quotes
+            if e[0] == "typography.symbols.curly_quotes":
+                continue
+            
+            # prepare error message
+            check = "check: %s, " %e[0] 
+            msg = "message: %s, " %e[1]
+            line = "line: %d, " %(1 + e[2])
+            column = "column: %d, " %(1 + e[3])
+            start = "start: %d, " %(1 + e[4])
+            end = "end: %d, " %(1 + e[5])
+            extent = "extent: %d, " %e[6]
+            severity = "severity: %s, " %e[7]
+            replace = "replacements: %s " %e[8]
+
+            # add errors to list
+            err_str = check + msg + line + column + start + end + extent + severity + replace
+            err_list.append(err_str)  
+
+        # display filename
+        print(filename)
+        
```

Comment:
I think this would be more user friendly if each error message included the file name and (if possible) the matched text, + a few words on either side of it. (Not the whole line, as that would end up being an entire paragraph.)

I'm thinking something like this:

```
sandwich-making.rst | ln 101 | col 23
...peanut butter, jelly and bread...
Use serial comma after penultimate item.
style-guide.serial-comma | warning
```

(I'm not sure what `start`, `stop`, and `extent` do, but I think they can be left out because everything the user needs to fix the problem is present there.)

Color would also help writers:

```
sandwich-making.rst | ln 101 | col 23      // blue
...peanut butter, jelly and bread...           // default color 
Use serial comma after penultimate item.    // yellow, orange, or red depending on severity
style-guide.serial-comma | warning    //  maybe gray
```

(Note: Feel free to use something else you like better, but -- I've used the [blessings](https://pypi.python.org/pypi/blessings/) for this in the past and found it very easy to work with.)

If the other (or some other) format for output is needed for circleCI, there can be like a `-verbose` flag or something.

Reply:
File name is displayed before the error list. Should the filename be displayed with each error?
 I will try to add the other features.

Reply:
start, stop and extent also tell the exact position of the error string counting word by word but that can be skipped safely.

Reply:
I will work on adding the features.

Reply:
@adammichaelwood Done!!
![screenshot from 2018-01-18 00-42-19](https://user-images.githubusercontent.com/15226041/35061848-7227e346-fbe8-11e7-94eb-158e0cf6ec2b.png)
These colors make me happy. This was a great idea indeed!!!